### DISCUSSION TITLE
QuestPDF with FontAwesome

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [QuestPDF/QuestPDF#321] [Q&A] [answered]
Hi all,

I'm trying to get QuestPDF to work with FontAwesome. I am running on Windows 10 Enterprise and targeting .NET 6. I have installed the FontAwesome OTF files and confirmed that they work correctly (word displays the glyphs). However, I cannot get them to work with QuestPDF.

Here is some sample code:
```cs
const string FontAwesomeLocationDot = "\uf3c5";
var document = Document.Create
(
    handler =>
    {
        handler.Page
        (
            page =>
            {
                page.Content()
                    .Text(FontAwesomeLocationDot)
                    .FontSize(20)
                    .FontFamily("Font Awesome 6 Pro")
                    .Light();
            }
        );
    }
);

await document.ShowInPreviewerAsync();
```
I would expect this to print out a location dot, however with light I get ~~T~~, regular gives me a lower case j, and bold gives me a bold lower case **j**.

This seems to happen regardless of the chosen glyph and I get the same behavior whether copy-pasting the glyph from the FontAwesome website or using the unicode escape as demonstrated above.

I have been able to get around it by doing this, but I have to guess at the correct scale and position to make the character look right. Additionally, I have to use `Unconstrained()` to avoid an exception on build from the glyph being too big (even after scaling).

```cs
.Unconstrained()
.Canvas(
    (canvas, size) =>
    {
        var svg = new SKSvg();
        svg.Load("./FontAwesomeGlyphs/location-dot.svg");
        
        // 0.035 is a little too big. Rounds down to 0.03 to make it look ok with the text.
        canvas.Scale((float)Math.Round(svg.Picture.CullRect.Height/size.Height, 2, MidpointRounding.ToZero));
        canvas.DrawPicture(svg.Picture, 50, 0, paint);
    });
```

Below is a screenshot of the two methods side by side (using a `.Row()`).
Column 1 uses the canvas renderer (code above). Offset by 50 to center the icon since this particular glyph is narrower than the others.
Column 2 uses the text method in the first example.
Column 3 is just the letter M for comparison with standard text.

![Comparison](https://media.discordapp.net/attachments/468748717684686868/1010373446569570324/unknown.png)

Edit: Props to #269 for showing the solution for loading an svg

**Comment by AverageCakeSlice** [comment]
I second this. Having support for some common SVG fonts would be incredible.

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Since this discussion was opened, QuestPDF has introduced native SVG support, which provides a much simpler and more reliable way to include icons like FontAwesome in your documents. You can find the documentation here:
https://www.questpdf.com/api-reference/svg.html

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
